src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -242,8 +253,10 @@ void StatefulWriter::send_any_unsent_changes()
 
             // Specific destination message group
             guids.at(0) = remoteReader->m_att.guid;
-            const LocatorList_t& locators = remoteReader->m_att.endpoint.remoteLocatorList;
-            RTPSMessageGroup group(mp_RTPSParticipant, this, RTPSMessageGroup::WRITER, m_cdrmessages, locators, guids);
+            const LocatorList_t remote_locators_shrinked{mp_RTPSParticipant->network_factory().ShrinkLocatorLists(
+                    {remoteReader->m_att.endpoint.remoteLocatorList})};
+            RTPSMessageGroup group(mp_RTPSParticipant, this, RTPSMessageGroup::WRITER, m_cdrmessages,
```

Comment:
If I remeber correctly, remoteLocatorList is shrinked inside `matched_reader_add`, so there is no need to shrink it here

Reply:
Decided by the team to keep in this way.